airflow/contrib/hooks/qubole_hook.py

```diff
@@ -0,0 +1,153 @@
+}
+
+class QuboleHook(BaseHook):
+    def __init__(self, *args, **kwargs):
+        conn = self.get_connection(kwargs['qubole_conn_id'])
+        Qubole.configure(api_token=conn.password, api_url=conn.host)
+        self.task_id = kwargs['task_id']
+        self.dag_id = kwargs['dag'].dag_id
+        self.kwargs = kwargs
+        self.args = self.create_cmd_args()
+        self.cls = COMMAND_CLASSES[self.kwargs['command_type']]
+        self.cmd = None
+
+    def execute(self):
+        args = self.cls.parse(self.args)
+        self.cmd = self.cls.create(**args)
+        logging.info("Qubole command created with Id: {0} and Status: {1}".format(str(self.cmd.id), self.cmd.status))
+
+        while not Command.is_done(self.cmd.status):
+            time.sleep(Qubole.poll_interval)
+            self.cmd = self.cls.find(self.cmd.id)
+            logging.info("Command Id: {0} and Status: {1}".format(str(self.cmd.id), self.cmd.status))
+
+        if self.kwargs.has_key('fetch_logs') and self.kwargs['fetch_logs'] == True:
+            logging.info("Logs for Command Id: {0}, {1}".format(str(self.cmd.id), self.cmd.get_log()))
+
+        if self.cmd.status != 'done':
+            raise AirflowException('Command Id: {0} failed with Status: {1}'.format(self.cmd.id, self.cmd.status))
+
+        return self.cmd.id
+
+    def kill(self, ti):
+        """
+        Kill (cancel) a Qubole commmand
+        :param ti: Task Instance of the dag, used to determine the Quboles command id
+        :return: response from Qubole
+        """
+        if self.cmd is None:
+            cmd_id = ti.xcom_pull(key="return_value", task_ids=self.task_id)
+            self.cmd = self.cls.find(cmd_id)
+        if self.cls and self.cmd:
+            logging.info('Sending KILL signal to Qubole Command Id: {0}'.format(self.cmd.id))
+            self.cmd.cancel()
+
+    def get_results(self, ti=None, fp=None, inline=True, delim=None, fetch=True):
+        """
+        Get results (or just s3 locations) of a command from Qubole and save it into a file
+        :param ti: Task Instance of the dag, used to determine the Quboles command id
+        :param fp: Optional file pointer, will create one and return if None passed
+        :param inline: True to download actual results, False to get s3 locations only
+        :param delim: Replaces the CTL-A chars with the given delim, defalt to ',' if None given
+        :param fetch: when inline is True, get results directly from s3 if results are large
+        :return: file location containing actual results or s3 locations of results
+        """
+        if fp is None:
+            iso = datetime.datetime.utcnow().isoformat()
+            logpath = os.path.expanduser(configuration.get('core', 'BASE_LOG_FOLDER'))
+            resultpath = logpath + '/' + self.dag_id + '/' + self.task_id + '/results'
+            configuration.mkdir_p(resultpath)
+            fp = open(resultpath + '/' + iso, 'wb')
```

Comment:
Yes, task_id will not be unique. `run_id` might be better option than timestamp because if number of dag runs are too much then it will be difficult for user to identify the correct results folder